contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/schema/DrillHiveTable.java

```diff
@@ -179,4 +179,9 @@ private void throwUnsupportedHiveDataTypeError(String hiveType) {
 
     throw new RuntimeException(errMsg.toString());
   }
+
+  @Override
+  public boolean providesDeferredSchema() {
```

Comment:
I think DrillHiveTable or other schema-aware table would return RelRecordType (in Calcite), when calling typeFactory.createStructType() as RowType.

For DrillDynamicTable, which is schema-less, the RowType would be RelDataTypeDrillImpl.